ironfish/src/account/accounts.ts

```diff
@@ -1297,6 +1303,75 @@ export class Accounts {
     return earliestHeader ? earliestHeader.hash : null
   }
 
+  async getLatestHeadHash(): Promise<Buffer | null> {
+    let latestHeader = null
+    for (const account of this.accounts.values()) {
+      const headHash = this.headHashes.get(account.id)
+
+      if (!headHash) {
+        continue
+      }
+
+      const header = await this.chain.getHeader(Buffer.from(headHash, 'hex'))
+
+      if (!header) {
+        // If no header is returned, the hash is likely invalid and we should remove it
+        await this.db.removeHeadHash(account)
+        continue
+      }
+
+      if (!latestHeader || latestHeader.sequence < header.sequence) {
+        latestHeader = header
+      }
+
+      // TODO: Check if any hashes are on known-forks
+    }
+
+    return latestHeader ? latestHeader.hash : null
+  }
+
+  updateAccountSyncStatus(): void {
+    const latestHash = this.chainProcessor.hash
+
+    for (const account of this.accounts.values()) {
+      const name = account.displayName
+      // If there is no loaded head hash, we assume all accounts should be updated
+      // as the chain is processed
+      if (!latestHash) {
```

Comment:
If I have one account up to date with a chain processor, import a new account (without the chain processor syncing more blocks, so the account doesn't have any notes stored), and run this method, will the `upToDate` be true?